### PR TITLE
feat: 🎸 Deprecate middleware in favour of middleware v2

### PR DESCRIPTION
**Pull request by prashantasdeveloper**
### Description

* Marks middleware v1 as deprecated.
* Adds checks to use middleware v2 queries if both v1 and v2 middleware are used to instantiate the SDK

### Breaking Changes

NA

### JIRA Link

DA-254

### Checklist

- [ ] Updated the Readme.md (if required) ?

### CODE REVIEW

**Review by polymath-eric — APPROVED** [review]
LGTM

**Review by VictorVicente — APPROVED** [review]

File: src/api/client/Polymesh.ts [hunk 1]
```diff
@@ -313,5 +316,12 @@ export class Polymesh {
   public get _middlewareApi(): ApolloClient<NormalizedCacheObject> {
     return this.context.middlewareApi;
   }
+
+  /**
+   * MiddlewareV2 client
+   */
+  public get _middlewareApiV2(): ApolloClient<NormalizedCacheObject> {
```
Comment (polymath-eric) [inline src/api/client/Polymesh.ts:323]: This should get a test

Comment (prashantasdeveloper) [reply src/api/client/Polymesh.ts:323]: The others are marked as   /* istanbul ignore next: not part of the official public API */
Should I mark here as well?

Comment (polymath-eric) [reply src/api/client/Polymesh.ts:324]: That works too.

I think a short one to add in `src/api/client/__tests__/Polymesh.ts` though.

**Review by polymath-eric — APPROVED** [review]

**Pull request merged**